### PR TITLE
Refactor apiserver and amazoncloudintegration render args into a struct

### PR DESCRIPTION
**Pull request by jaderhs**
## Description

This change is a small improvement to change apiserver and amazoncloudintegration render args into a struct
This change will make it easier to add new render parameter without the need to update ~100 function calls in the tests that will not use the new parameter

This change affects the apiServer and amazonCloudIntegration renderers.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/amazoncloudintegration_test.go [hunk 1]
```diff
@@ -74,7 +81,14 @@ var _ = Describe("AmazonCloudIntegration rendering tests", func() {
 
 	It("should render an AmazonCloudConfiguration with specified configuration", func() {
 		// AmazonCloudIntegration(aci *operatorv1.AmazonCloudIntegration, installation *operator.Installation, cred *AmazonCredential, ps []*corev1.Secret, openshift bool) (Component, error) {
-		component, err := render.AmazonCloudIntegration(instance, installation, credential, nil, openshift)
+		cfg := &render.AmazonCloudIntegrationConfiguration{
```
Comment (tmjd) [inline pkg/render/amazoncloudintegration_test.go:84]: Similar to my comment for apiserver, I think you should have a cfg be created in the BeforeEach so the tests can use defaults and only need to set any specific fields they need to tweak.

**Review by tmjd — COMMENTED** [review]
Looking good, I think a few tweaks to the test updates would be good to take advantage of the config struct.

File: pkg/render/apiserver_test.go [hunk 2]
```diff
@@ -99,7 +99,13 @@ var _ = Describe("API server rendering tests (Calico Enterprise)", func() {
 		}
 
 		// APIServer(registry string, tlsKeyPair *corev1.Secret, pullSecrets []*corev1.Secret, openshift bool
-		component, err := render.APIServer(k8sServiceEp, instance, false, nil, nil, nil, nil, nil, openshift, nil, clusterDomain)
+		cfg := &render.APIServerConfiguration{
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/apiserver_test.go]: Moved cfg to BeforeEach section

File: pkg/render/amazoncloudintegration_test.go (hunk 1 above)
Comment (jaderhs) [reply pkg/render/amazoncloudintegration_test.go]: Moved cfg to BeforeEach section

File: pkg/render/amazoncloudintegration_test.go [hunk 3]
```diff
@@ -62,9 +62,16 @@ var _ = Describe("AmazonCloudIntegration rendering tests", func() {
...
-		component, err := render.AmazonCloudIntegration(instance, &operatorv1.InstallationSpec{
-			ControlPlaneNodeSelector: map[string]string{"foo": "bar"},
-		}, credential, nil, false)
+		cfg := &render.AmazonCloudIntegrationConfiguration{
+			AmazonCloudIntegration: instance,
+			Installation: &operatorv1.InstallationSpec{
+				ControlPlaneNodeSelector: map[string]string{"foo": "bar"},
+			},
+			Credentials: credential,
+			PullSecrets: nil,
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/amazoncloudintegration_test.go]: Left default fields off. Setting those inside tests tha will need them

**Review by tmjd — APPROVED** [review]
LGTM

File: pkg/render/apiserver.go [hunk 4]
```diff
@@ -108,69 +100,68 @@ func APIServer(k8sServiceEndpoint k8sapi.ServiceEndpoint,
...
+		tlsSecrets:     tlsSecrets,
+		tlsAnnotations: tlsHashAnnotations,
 	}, nil
 }
 
+// APIServerConfiguration contains all the config information apiServerComponent needs to render component.
+type APIServerConfiguration struct {
+	K8SServiceEndpoint          k8sapi.ServiceEndpoint
+	Installation                *operatorv1.InstallationSpec
+	HostNetwork                 bool
```
Comment (tmjd) [inline pkg/render/apiserver.go:136]: Please switch this to ForceHostNetwork, because HostNetwork may still be used even if this field is not set.

**Review by tmjd — APPROVED** [review]
LGTM

File: pkg/render/apiserver.go (hunk 4 above)
Comment (jaderhs) [reply pkg/render/apiserver.go]: Done

**Pull request merged**